File: repositori_SSGG/scripts/utils/templates/code_snippets/PipeTA.pl
#!/usr/bin/env perl
########################################################
# Copyright 2012 - Sistemas Genómicos S.L.
# @Desc: Script for launching pipelines to analyze
# @Author: Arbol
# @Contributors: Arbol
########################################################

# -----------------------
# -- Include libraries --
# -----------------------
use File::Basename;
use Cwd;
use Cwd qw( abs_path );
use strict;
use warnings;
use POSIX qw( strftime );
use JSON -support_by_pp;
use Data::Dumper;
use Getopt::Long;

# ---------------
# -- functions --
# ---------------

sub usage ();

# -----------------------
# -- global parameters --
# -----------------------
#my $fooGlobalVar = $ENV{"PATH"};

# ------------------------
# -- default parameters --
# ------------------------
my $conf_file = "/share/apps/etc/pipelines/config.json";

&main();

sub main (){
	# Get input parameters
	GetOptions (	
					"c=s"	=> \$conf_file,
	);
	
	# Display name and version, and check input arguments
	(my $basename, my $dir, my $ext) = fileparse($0, qr/\.[^.]*/);
	print "$basename$ext v0.1
	Script for launching analysis pipelines (PIPElines To Analyze->PipeTA). It loads configurations, paths and folders, so that it assures the use of the right information. · Copyright 2012 by Arbol 
	***************************************************************************************************************************************************\n";
	if (not defined $conf_file) { usage; }
	
	# Load default configuration from XML file:
	my $json;
	{
		local $/; #Enable 'slurp' mode
		open my $fh, "<", $conf_file;
		$json = <$fh>;
		close $fh;
	}
	my $data = decode_json($json);
	print Dumper($data); #uncomment for showing contents of configuration stored in $data
}

sub usage () {
# Displays the script usage
	# Parameters:
	my ( $scriptname ) = @_;

	print "
Usage:
  $scriptname <options>

* Input (mandatory parameters): *

* Options: *
-c Configuration_file    File containing all the configuration parameters that must be loaded (default /share/apps/etc/pipelines/config.json)
* Output: *

* Examples of running: *
 \$$scriptname 
 \n";
 
	exit(1);
}

exit;

File: repositori_SSGG/scripts/utils/templates/code_snippets/PipeTA.py
#!/usr/bin/env python
"""
SYNOPSIS
    PipeTA.py [-h,--help] [-v,--verbose] [--version] [-c,--config]
DESCRIPTION
    PipeTA: Pipeline To Analyze, a script for retrieving general configuration parameters for launching pipelines
EXAMPLES
    PipeTA.py
EXIT STATUS
    TODO: List exit codes
AUTHOR
    Arbol <oscar.rodriguez@sistemasgenomicos.com>
LICENSE
    This script belongs to Sistemas Genomicos S.L.
VERSION
    PipeTA.py v0.1.0
"""

import sys, os, traceback, optparse
import time
import re
import json
import os
#from pexpect import run, spawn

def main ():

    global options, args

    # ****************************** main body *********************************
    print 'PipeTA: Pipeline To Analyze, a script for retrieving general configuration parameters for launching pipelines'
    
    config = json.load(open(options.config))
    
    #print(os.environ)
    os.environ["capture"] = config["references"]["panels"][analysis]["capture"]
    os.environ["target"] = config["references"]["panels"][analysis]["target"]
    os.environ["involved_chrs"] = config["references"]["panels"][analysis]["involved_chrs"]
    
    
    print(os.environ)
    
if __name__ == '__main__':
    try:
        start_time = time.time()
        parser = optparse.OptionParser(formatter=optparse.TitledHelpFormatter(), usage=globals()['__doc__'], version='0.1.0')
        parser.add_option ('-v', '--verbose', action='store_true', default=False, help='verbose output')
        parser.add_option ('-c', '--config', action='store_true', dest="config", default="/share/apps/etc/pipelines/config.json", help='Json configuration file')
        (options, args) = parser.parse_args()
        if len(args) < 1:
            parser.error ('missing argument')
        else: analysis = args[0]
        if options.verbose: print time.asctime()
        main()
        if options.verbose: print time.asctime()
        if options.verbose: print 'Duration of script run:',
        if options.verbose: print (time.time() - start_time) / 60.0
        sys.exit(0)
    except KeyboardInterrupt, e: # Ctrl-C
        raise e
    except SystemExit, e: # sys.exit()
        raise e
    except Exception, e:
        print 'ERROR, UNEXPECTED EXCEPTION'
        print str(e)
        traceback.print_exc()
        os._exit(1)
